### DISCUSSION TITLE
Support for Unraid?

### DISCUSSION THREAD

**Discussion by vkux** [Taxel/PlexTraktSync#96] [Q&A] [answered]
I currently run your application as part of my Windows PMS. I plan to migrate to an Unraid-based system. Would it be possible to host this as a docker or as part of the Community Applications in Unraid?

I'm a newbie and a non-techy unfortunately, otherwise I would've offered my help. :(

**Comment by simonc56** [comment]
The official docker version is currently under construction -> #88 
Stay tuned.

**Reply by heezy01** [reply]
need some definitive help on how to set this up in Unraid. i ran, "docker pull ghcr.io/taxel/plextraktsync:main" in the terminal window.

docker pull ghcr.io/taxel/plextraktsync:main
main: Pulling from taxel/plextraktsync
4e9f2cdf4387: Pull complete 
80d1de1f04a6: Pull complete 
ae4d38e8e7e2: Pull complete 
b926f202311d: Pull complete 
94011579eee9: Pull complete 
9ae4800878d5: Pull complete 
3c450898ac20: Pull complete 
0e580efbd823: Pull complete 
5d5c62529f0f: Pull complete 
Digest: sha256:d42f694dd3ae2fa0384544d5e49583f99426354a056c2512d123f609827d8d59
Status: Downloaded newer image for ghcr.io/taxel/plextraktsync:main
ghcr.io/taxel/plextraktsync:main


then i go to the Docker tab, but there is no plextraktsync docker. i enable advanced view and plextraktsync shows up as an orphan image and it is currently stopped. only command to run is to remove it.

any help to properly install the plextraktsync docker in Unraid is greatly appreciated.

**Reply by glensc** [reply]
no idea about unraid, but don't pull "main" branch, it's a development version, use "latest" or a specific tag. where did you take info to use "main" branch? please tell and shoot the person suggesting it.

**Reply by heezy01** [reply]
i got the latest installed, but it still ends up as an orphan image. i just went ahead and installed the script the pipx way. after adjusting the config file to my liking, i was able to create a user script to run plextraktsync every 2 hours with cron.

now i just need to figure out a bash script to run "plextraktsync watch" in the background automatically.

**Reply by glensc** [reply]
perhaps you submit pull request to update README on how to set up the script on Unraid for other users to understand too.

**Reply by fnwc** [reply]
@heezy01 I added a pull request [here](https://github.com/Taxel/PlexTraktSync/pull/590/files) if you want to see how to set it up in Unraid.

**Reply by heezy01** [reply]
> @heezy01 I added a pull request [here](https://github.com/Taxel/PlexTraktSync/pull/590/files) if you want to see how to set it up in Unraid.

thanks for this. i'll give it a try sometime. with the pipx install, i couldn't figure out how to run the watch command in the background from a script. so i just installed the github way and it seems to be working for me right now.

**Reply by glensc** [reply]
if installed with pipx, the command should be `plextraktsync watch`.

**Comment by simonc56** [comment, marked as answer]
It is hosted as a docker container:
- https://github.com/Taxel/PlexTraktSync/pkgs/container/plextraktsync/44009987?tag=latest

note: use `latest`, not `main` (`main` is development version).

**Reply by fnwc** [reply]
The main issue I'm running into is that there's no way that I can see to authenticate the users needed using the python3 method described. Looking at the log I see "Please enter your Plex username.

![image](https://user-images.githubusercontent.com/7389518/140012039-7f6f3a12-6a48-46d1-99a7-25da8985000e.png)

However, due to the way the docker is set up in Unraid (I'm guessing), it's not a listed module using something like help("modules"). I'm not really that familiar with docker, but there doesn't appear to be any variables that I can set to set my authorization for either Plex and/or Trakt.

**Reply by glensc** [reply]
You've written your question in a way, who can answer only someone who has something called "Unraid". I'm absolutely unfamiliar with it.

You need to run the container in interactive mode to be able to input answers to questions it asks.

Try to present your problem in a reproducible way, then maybe I can answer.

**Reply by fnwc** [reply]
I totally get it. The situation is unique to Unraid because the docker system isn't registered as a module that python can see, I'm guessing.
Is there any way to generate the appropriate credentials a different way, either through manual editing of config files or exposing a docker variable the appropriate information?

**Reply by simonc56** [reply]
You can generate credential files (`.env` and `.pytrakt.json`) with any desktop computer where you can run the script in a standard console. Then copy/paste those files to your docker config folder.

**Comment by xiongster** [comment]
Thanks for this docker container build. Can anybody please specify how to run this im unraid?

**Reply by fnwc** [reply]
@xiongster I added a pull request [here](https://github.com/Taxel/PlexTraktSync/pull/590/files) if you want to see how to set it up in Unraid.

**Reply by heezy01** [reply]
i got it up and running through a docker container in unraid, but i need to edit the config.json file. i don't know where the config file is located. i run the info command and the config folder is located in, /app/config. where is that folder in unraid?

**Reply by fnwc** [reply]
Open a terminal in the container itself. It will be in the root directory. You can’t see it in the Unraid file system because that’s how containers work.

**Reply by xiongster** [reply]
@fnwc  Thanks bro. Can you please tell how to automate this? Like on windows I had option to run sync everytime at startup.

**Reply by neo2zero** [reply]
Thanks to everyone who's been working on this. Much appreciated.

That said, I installed this on unRaid using the instructions provided. After supplying my Plex and Trakt information, it started the initial sync.

Fast forward a couple of days, I have acquired a few tv shows and movies,  as well as watching a bit of both.

I start the docker, (as it stops itself after completing a sync, I assume this is standard behavior) and plextraktsync starts the sync process. It sees that I have newly acquired movies and tv, and updates Trakt accordingly, EXCEPT it does not update the watch history for either tv or movies.

I run plextraktsync again however, it still doesn't update watch history from the initial sync. If the initial sync worked to update watch history, why wouldn't subsequent sync update watch history?
